lib/route-helpers.ts

```diff
@@ -50,3 +50,34 @@ export function generateInitResponseBody(event: Record<string, any>): initRespon
     heartbeatInterval: event.heartbeatInterval || process.env.HEARTBEAT_INTERVAL || 5000,
   };
 }
+
+function shouldValidateOrigin(): boolean {
+  return (process.env.CORS_ORIGIN !== undefined);
+}
+
+function getAllowedOrigins(): string[] {
+  if (process.env.CORS_ORIGIN) {
+    return process.env.CORS_ORIGIN.split(",").map(o => o.trim());
+  }
+  return [];
+}
+
+function isAllowedOrigin(origin: string):  boolean {
+  if (shouldValidateOrigin()) {
+    const allowedOrigins = getAllowedOrigins();
+    return allowedOrigins.includes(origin);
+  }
+  return true;
+}
+
+export function generateResponseHeaders(origin: string) {
+  let responseHeaders = defaultResponseHeaders;
+
+  if (shouldValidateOrigin()) {
+    if (isAllowedOrigin(origin)) {
+      responseHeaders['Access-Control-Allow-Origin'] = origin;
+      responseHeaders['Vary'] = 'Origin';
+    }
+    return responseHeaders;
+  }
```

Comment:
If this event sink endpoint will at some point be handling a lot of incoming requests, I think there are some optimizations that could be made through reducing function calls, accessors, caching.

I think you could get rid of the `shouldValidateOrigin` method by returning `true` in `isAllowedOrigin` if `getAllowedOrigins().length === 0`.

I might be wrong, but I assume the `process.env` isn't going to change while the server is running, so the `CORS_ORIGIN` variable could be fetched and parsed once `const allowedOrigins = process.env.CORS_ORIGIN.split(",").map(o => o.trim());`. It would get rid of the relatively expensive parsing of `CORS_ORIGIN` that now happens on each request.

It would also be a _minor_ CPU time improvement if the empty array length check is cached as a boolean `const allOriginsAllowed = allowedOrigins.length;`.

```suggestion
const allowedOrigins = process.env.CORS_ORIGIN?.split(",").map(o => o.trim());
const allowAnyOrigin = allowedOrigins.length === 0 || undefined;

const isAllowedOrigin = (origin: string) => allowedOrigins.includes(origin);

export function generateResponseHeaders(origin: string) {
  let responseHeaders = defaultResponseHeaders;

  if (allowAnyOrigin || isAllowedOrigin(origin)) {
    responseHeaders['Access-Control-Allow-Origin'] = origin;
    responseHeaders['Vary'] = 'Origin';
  }
    return responseHeaders;
  }
```

Reply:
Thank you, I addressed some of these and updated the PR.
I kept the process.env reading inside the function scope as the entire module is not fully enclosed. It could vary if is used in a Lambda vs fastify context perhaps.